nixos/modules/services/logging/logrotate.nix

```diff
@@ -103,6 +103,13 @@ let
     )
   );
 
+  mailOption =
+    # add mail option to service if a mail is requested in config
+    # this ugly match will be replaced by cleaner attribute check in
+    # the near future
+    if builtins.match "(.*[[:space:]])?mail[[:space:]].*" (lib.fileContents configFile) != null
```

Comment:
I **think** this is an [IFD](https://nixos.wiki/wiki/Import_From_Derivation), because `fileContents` (a `readFile` in disguise) is applied on `configFile` (a `writeText` derivation in disguise). [But](https://github.com/NixOS/nixpkgs/pull/101071#issuecomment-736779044):
> We cannot use IFD in NixOS/Nixpkgs. It's disabled on hydra.nixos.org to enforce this,
> not just because it's bad for Hydra but because it's bad for users
> (e.g. just evaluating a configuration could trigger an arbitrary amount of building).

Reply:
@martined, if confirmed, you should be able to workaround that [by using runCommand](https://github.com/ju1m/nixpkgs/pull/12/commits/08106b7fe5074e53fedfa55cb9b9a6dcef0b50c9).

Reply:
That's it, it's an IFD:
```
[nix-shell:~/.cache/nixpkgs-review/pr-162225/nixpkgs]$ nix --no-allow-import-from-derivation -L build -f nixos/tests/logrotate.nix --show-trace
error: cannot build '/nix/store/67qkff25lz2gw8wmpdpasrha4rw3f14i-logrotate.conf.drv' during evaluation because the option 'allow-import-from-derivation' is disabled

       … while realising the context of path '/nix/store/vmlxlwsq5jwvxxynfi5q3j2317x5sydg-logrotate.conf'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/strings.nix:744:43:

          743|   */
          744|   fileContents = file: removeSuffix "\n" (readFile file);
             |                                           ^
          745|

       … while evaluating 'removeSuffix'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/strings.nix:486:5:

          485|     # Input string
          486|     str:
             |     ^
          487|     let

       … from call site

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/strings.nix:744:24:

          743|   */
          744|   fileContents = file: removeSuffix "\n" (readFile file);
             |                        ^
          745|

       … while evaluating 'fileContents'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/strings.nix:744:18:

          743|   */
          744|   fileContents = file: removeSuffix "\n" (readFile file);
             |                  ^
          745|

       … from call site

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/nixos/modules/services/logging/logrotate.nix:110:60:

          109|     # the near future
          110|     if builtins.match "(.*[[:space:]])?mail[[:space:]].*" (lib.fileContents configFile) != null
             |                                                            ^
          111|     then "--mail=${pkgs.mailutils}/bin/mail"

       … while evaluating the attribute 'value'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/types.nix:424:60:

          423|           # Push down position info.
          424|           (map (def: mapAttrs (n: v: { inherit (def) file; value = v; }) def.value) defs)));
             |                                                            ^
          425|       emptyValue = { value = {}; };

       … while evaluating 'dischargeProperties'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/modules.nix:699:25:

          698|   */
          699|   dischargeProperties = def:
             |                         ^
          700|     if def._type or "" == "merge" then

       … from call site

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/modules.nix:628:137:

          627|         defs' = concatMap (m:
          628|           map (value: { inherit (m) file; inherit value; }) (builtins.addErrorContext "while evaluating definitions from `${m.file}':" (dischargeProperties m.value))
             |                                                                                                                                         ^
          629|         ) defs;

       … while evaluating definitions from `/home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/nixos/modules/services/logging/logrotate.nix':

       … while evaluating anonymous lambda

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/modules.nix:627:28:

          626|         # Process mkMerge and mkIf properties.
          627|         defs' = concatMap (m:
             |                            ^
          628|           map (value: { inherit (m) file; inherit value; }) (builtins.addErrorContext "while evaluating definitions from `${m.file}':" (dischargeProperties m.value))

       … from call site

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/modules.nix:627:17:

          626|         # Process mkMerge and mkIf properties.
          627|         defs' = concatMap (m:
             |                 ^
          628|           map (value: { inherit (m) file; inherit value; }) (builtins.addErrorContext "while evaluating definitions from `${m.file}':" (dischargeProperties m.value))

       … while evaluating the attribute 'values'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/modules.nix:740:7:

          739|     in {
          740|       values = concatMap (def: if getPrio def == highestPrio then [(strip def)] else []) defs;
             |       ^
          741|       inherit highestPrio;

       … while evaluating the attribute 'values'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/modules.nix:641:9:

          640|       in {
          641|         values = defs''';
             |         ^
          642|         inherit (defs'') highestPrio;

       … while evaluating the attribute 'optionalValue'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/modules.nix:659:5:

          658|
          659|     optionalValue =
             |     ^
          660|       if isDefined then { value = mergedValue; }

       … while evaluating anonymous lambda

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/types.nix:420:86:

          419|       merge = loc: defs:
          420|         mapAttrs (n: v: v.value) (filterAttrs (n: v: v ? value) (zipAttrsWith (name: defs:
             |                                                                                      ^
          421|             (mergeDefinitions (loc ++ [name]) elemType defs).optionalValue

       … from call site

       … while evaluating anonymous lambda

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/types.nix:420:51:

          419|       merge = loc: defs:
          420|         mapAttrs (n: v: v.value) (filterAttrs (n: v: v ? value) (zipAttrsWith (name: defs:
             |                                                   ^
          421|             (mergeDefinitions (loc ++ [name]) elemType defs).optionalValue

       … from call site

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/attrsets.nix:128:62:

          127|   filterAttrs = pred: set:
          128|     listToAttrs (concatMap (name: let v = set.${name}; in if pred name v then [(nameValuePair name v)] else []) (attrNames set));
             |                                                              ^
          129|

       … while evaluating anonymous lambda

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/attrsets.nix:128:29:

          127|   filterAttrs = pred: set:
          128|     listToAttrs (concatMap (name: let v = set.${name}; in if pred name v then [(nameValuePair name v)] else []) (attrNames set));
             |                             ^
          129|

       … from call site

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/attrsets.nix:128:18:

          127|   filterAttrs = pred: set:
          128|     listToAttrs (concatMap (name: let v = set.${name}; in if pred name v then [(nameValuePair name v)] else []) (attrNames set));
             |                  ^
          129|

       … while evaluating 'filterAttrs'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/attrsets.nix:127:23:

          126|   */
          127|   filterAttrs = pred: set:
             |                       ^
          128|     listToAttrs (concatMap (name: let v = set.${name}; in if pred name v then [(nameValuePair name v)] else []) (attrNames set));

       … from call site

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/types.nix:420:35:

          419|       merge = loc: defs:
          420|         mapAttrs (n: v: v.value) (filterAttrs (n: v: v ? value) (zipAttrsWith (name: defs:
             |                                   ^
          421|             (mergeDefinitions (loc ++ [name]) elemType defs).optionalValue

       … while evaluating 'merge'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/types.nix:419:20:

          418|       check = isAttrs;
          419|       merge = loc: defs:
             |                    ^
          420|         mapAttrs (n: v: v.value) (filterAttrs (n: v: v ? value) (zipAttrsWith (name: defs:

       … from call site

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/modules.nix:649:59:

          648|       if isDefined then
          649|         if all (def: type.check def.value) defsFinal then type.merge loc defsFinal
             |                                                           ^
          650|         else let allInvalid = filter (def: ! type.check def.value) defsFinal;

       … while evaluating the attribute 'mergedValue'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/modules.nix:647:5:

          646|     # Type-check the remaining definitions, and merge them. Or throw if no definitions.
          647|     mergedValue =
             |     ^
          648|       if isDefined then

       … while evaluating the option `systemd.services.logrotate.serviceConfig':

       … while evaluating the attribute 'value'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/modules.nix:613:9:

          612|     in warnDeprecation opt //
          613|       { value = builtins.addErrorContext "while evaluating the option `${showOption loc}':" value;
             |         ^
          614|         inherit (res.defsFinal') highestPrio;

       … while evaluating anonymous lambda

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/modules.nix:192:72:

          191|           # For definitions that have an associated option
          192|           declaredConfig = mapAttrsRecursiveCond (v: ! isOption v) (_: v: v.value) options;
             |                                                                        ^
          193|

       … from call site

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/attrsets.nix:304:20:

          303|               then recurse (path ++ [name]) value
          304|               else f (path ++ [name]) value;
             |                    ^
          305|         in mapAttrs g;

       … while evaluating 'g'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/attrsets.nix:301:19:

          300|           g =
          301|             name: value:
             |                   ^
          302|             if isAttrs value && cond value

       … from call site

       … while evaluating 'optional'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/lists.nix:254:20:

          253|   */
          254|   optional = cond: elem: if cond then [elem] else [];
             |                    ^
          255|

       … from call site

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/nixos/modules/system/boot/systemd.nix:916:16:

          915|             concatLists [
          916|               (optional (type == "oneshot" && (restart == "always" || restart == "on-success"))
             |                ^
          917|                 "Service '${name}.service' with 'Type=oneshot' cannot have 'Restart=always' or 'Restart=on-success'"

       … while evaluating anonymous lambda

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/nixos/modules/system/boot/systemd.nix:909:16:

          908|       mapAttrsToList
          909|         (name: service:
             |                ^
          910|           let

       … from call site

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/attrsets.nix:259:16:

          258|   mapAttrsToList = f: attrs:
          259|     map (name: f name attrs.${name}) (attrNames attrs);
             |                ^
          260|

       … while evaluating anonymous lambda

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/attrsets.nix:259:10:

          258|   mapAttrsToList = f: attrs:
          259|     map (name: f name attrs.${name}) (attrNames attrs);
             |          ^
          260|

       … from call site

       … while evaluating the attribute 'value'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/modules.nix:475:44:

          474|       defnsByName' = byName "config" (module: value:
          475|           [{ inherit (module) file; inherit value; }]
             |                                            ^
          476|         ) configs;

       … while evaluating 'dischargeProperties'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/modules.nix:699:25:

          698|   */
          699|   dischargeProperties = def:
             |                         ^
          700|     if def._type or "" == "merge" then

       … from call site

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/modules.nix:628:137:

          627|         defs' = concatMap (m:
          628|           map (value: { inherit (m) file; inherit value; }) (builtins.addErrorContext "while evaluating definitions from `${m.file}':" (dischargeProperties m.value))
             |                                                                                                                                         ^
          629|         ) defs;

       … while evaluating definitions from `/home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/nixos/modules/system/boot/systemd.nix':

       … while evaluating anonymous lambda

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/modules.nix:627:28:

          626|         # Process mkMerge and mkIf properties.
          627|         defs' = concatMap (m:
             |                            ^
          628|           map (value: { inherit (m) file; inherit value; }) (builtins.addErrorContext "while evaluating definitions from `${m.file}':" (dischargeProperties m.value))

       … from call site

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/modules.nix:627:17:

          626|         # Process mkMerge and mkIf properties.
          627|         defs' = concatMap (m:
             |                 ^
          628|           map (value: { inherit (m) file; inherit value; }) (builtins.addErrorContext "while evaluating definitions from `${m.file}':" (dischargeProperties m.value))

       … while evaluating the attribute 'values'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/modules.nix:740:7:

          739|     in {
          740|       values = concatMap (def: if getPrio def == highestPrio then [(strip def)] else []) defs;
             |       ^
          741|       inherit highestPrio;

       … while evaluating the attribute 'values'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/modules.nix:641:9:

          640|       in {
          641|         values = defs''';
             |         ^
          642|         inherit (defs'') highestPrio;

       … while evaluating the attribute 'mergedValue'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/modules.nix:647:5:

          646|     # Type-check the remaining definitions, and merge them. Or throw if no definitions.
          647|     mergedValue =
             |     ^
          648|       if isDefined then

       … while evaluating the option `warnings':

       … while evaluating the attribute 'value'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/modules.nix:613:9:

          612|     in warnDeprecation opt //
          613|       { value = builtins.addErrorContext "while evaluating the option `${showOption loc}':" value;
             |         ^
          614|         inherit (res.defsFinal') highestPrio;

       … while evaluating anonymous lambda

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/modules.nix:192:72:

          191|           # For definitions that have an associated option
          192|           declaredConfig = mapAttrsRecursiveCond (v: ! isOption v) (_: v: v.value) options;
             |                                                                        ^
          193|

       … from call site

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/attrsets.nix:304:20:

          303|               then recurse (path ++ [name]) value
          304|               else f (path ++ [name]) value;
             |                    ^
          305|         in mapAttrs g;

       … while evaluating 'g'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/attrsets.nix:301:19:

          300|           g =
          301|             name: value:
             |                   ^
          302|             if isAttrs value && cond value

       … from call site

       … while evaluating 'fold''

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/lists.nix:55:15:

           54|       len = length list;
           55|       fold' = n:
             |               ^
           56|         if n == len

       … from call site

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/lists.nix:59:8:

           58|         else op (elemAt list n) (fold' (n + 1));
           59|     in fold' 0;
             |        ^
           60|

       … while evaluating 'foldr'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/lists.nix:52:20:

           51|   */
           52|   foldr = op: nul: list:
             |                    ^
           53|     let

       … from call site

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/trivial.nix:369:33:

          368|
          369|   showWarnings = warnings: res: lib.foldr (w: x: warn w x) res warnings;
             |                                 ^
          370|

       … while evaluating 'showWarnings'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/trivial.nix:369:28:

          368|
          369|   showWarnings = warnings: res: lib.foldr (w: x: warn w x) res warnings;
             |                            ^
          370|

       … from call site

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/nixos/modules/system/activation/top-level.nix:129:10:

          128|     then throw "\nFailed assertions:\n${concatStringsSep "\n" (map (x: "- ${x}") failedAssertions)}"
          129|     else showWarnings config.warnings baseSystem;
             |          ^
          130|

       … while evaluating 'fold''

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/lists.nix:55:15:

           54|       len = length list;
           55|       fold' = n:
             |               ^
           56|         if n == len

       … from call site

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/lists.nix:59:8:

           58|         else op (elemAt list n) (fold' (n + 1));
           59|     in fold' 0;
             |        ^
           60|

       … while evaluating 'foldr'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/lists.nix:52:20:

           51|   */
           52|   foldr = op: nul: list:
             |                    ^
           53|     let

       … from call site

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/nixos/modules/system/activation/top-level.nix:132:12:

          131|   # Replace runtime dependencies
          132|   system = foldr ({ oldDependency, newDependency }: drv:
             |            ^
          133|       pkgs.replaceDependency { inherit oldDependency newDependency drv; }

       … while evaluating the attribute 'value'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/modules.nix:475:44:

          474|       defnsByName' = byName "config" (module: value:
          475|           [{ inherit (module) file; inherit value; }]
             |                                            ^
          476|         ) configs;

       … while evaluating 'dischargeProperties'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/modules.nix:699:25:

          698|   */
          699|   dischargeProperties = def:
             |                         ^
          700|     if def._type or "" == "merge" then

       … from call site

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/modules.nix:628:137:

          627|         defs' = concatMap (m:
          628|           map (value: { inherit (m) file; inherit value; }) (builtins.addErrorContext "while evaluating definitions from `${m.file}':" (dischargeProperties m.value))
             |                                                                                                                                         ^
          629|         ) defs;

       … while evaluating definitions from `/home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/nixos/modules/system/activation/top-level.nix':

       … while evaluating anonymous lambda

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/modules.nix:627:28:

          626|         # Process mkMerge and mkIf properties.
          627|         defs' = concatMap (m:
             |                            ^
          628|           map (value: { inherit (m) file; inherit value; }) (builtins.addErrorContext "while evaluating definitions from `${m.file}':" (dischargeProperties m.value))

       … from call site

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/modules.nix:627:17:

          626|         # Process mkMerge and mkIf properties.
          627|         defs' = concatMap (m:
             |                 ^
          628|           map (value: { inherit (m) file; inherit value; }) (builtins.addErrorContext "while evaluating definitions from `${m.file}':" (dischargeProperties m.value))

       … while evaluating the attribute 'values'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/modules.nix:740:7:

          739|     in {
          740|       values = concatMap (def: if getPrio def == highestPrio then [(strip def)] else []) defs;
             |       ^
          741|       inherit highestPrio;

       … while evaluating the attribute 'values'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/modules.nix:641:9:

          640|       in {
          641|         values = defs''';
             |         ^
          642|         inherit (defs'') highestPrio;

       … while evaluating the attribute 'mergedValue'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/modules.nix:647:5:

          646|     # Type-check the remaining definitions, and merge them. Or throw if no definitions.
          647|     mergedValue =
             |     ^
          648|       if isDefined then

       … while evaluating the option `system.build.toplevel':

       … while evaluating the attribute 'value'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/modules.nix:613:9:

          612|     in warnDeprecation opt //
          613|       { value = builtins.addErrorContext "while evaluating the option `${showOption loc}':" value;
             |         ^
          614|         inherit (res.defsFinal') highestPrio;

       … while evaluating anonymous lambda

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/modules.nix:192:72:

          191|           # For definitions that have an associated option
          192|           declaredConfig = mapAttrsRecursiveCond (v: ! isOption v) (_: v: v.value) options;
             |                                                                        ^
          193|

       … from call site

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/attrsets.nix:304:20:

          303|               then recurse (path ++ [name]) value
          304|               else f (path ++ [name]) value;
             |                    ^
          305|         in mapAttrs g;

       … while evaluating 'g'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/attrsets.nix:301:19:

          300|           g =
          301|             name: value:
             |                   ^
          302|             if isAttrs value && cond value

       … from call site

       … while evaluating anonymous lambda

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/lib/attrsets.nix:421:24:

          420|     let f = attrPath:
          421|       zipAttrsWith (n: values:
             |                        ^
          422|         let here = attrPath ++ [n]; in

       … from call site

       … while evaluating the attribute 'buildCommand' of the derivation 'nixos-vm'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:205:7:

          204|     // (lib.optionalAttrs (attrs ? name || (attrs ? pname && attrs ? version)) {
          205|       name =
             |       ^
          206|         let

       … while evaluating the attribute 'buildCommand' of the derivation 'nixos-test-driver-logrotate'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:205:7:

          204|     // (lib.optionalAttrs (attrs ? name || (attrs ? pname && attrs ? version)) {
          205|       name =
             |       ^
          206|         let

       … while evaluating the attribute 'buildCommand' of the derivation 'vm-test-run-logrotate'

       at /home/julm/.cache/nixpkgs-review/pr-162225/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:205:7:

          204|     // (lib.optionalAttrs (attrs ? name || (attrs ? pname && attrs ? version)) {
          205|       name =
             |       ^
          206|         let
```

Reply:
hmpf, yes it's a readFile. I knew it was ugly but I see it used in a few place so I didn't think it was forbidden...
I've worked around this by storing the file content in an intermediate `configText` variable instead, which is used both when writing file and for the match.

Reply:
Indeed, much simpler ^^